jbpm-task-assigning/jbpm-task-assigning-core/src/main/java/org/jbpm/task/assigning/model/Task.java

```diff
@@ -0,0 +1,333 @@
+ * With the declaration below a shadow variable is defined for keeping a reference to the anchor. This shadow variable
+ * is populated and kept consistent by the solver.
+ * <p>
+ * Shadow variable:
+ * Let all Tasks have a reference to the anchor of the chain, the assigned user.
+ * @AnchorShadowVariable(sourceVariableName = "previousTaskOrUser")
+ * private User user;
+ * <p>
+ * CustomShadowVariable startTime:
+ * Additionally is a convenient shadow variable is declared for having the startTime of a task already calculated.
+ * @CustomShadowVariable(variableListenerClass = StartTimeUpdatingVariableListener.class,
+ * // Arguable, to adhere to API specs (although this works), nextTask and user should also be a source,
+ * // because this shadow must be triggered after nextTask and user (but there is no need to be triggered by those)
+ * sources = {@PlanningVariableReference(variableName = "previousTaskOrUser")})
+ * private Integer startTime;
+ * <p>
+ * So the variableListenerClass is invoked when the source variable is changed/assigned.
+ */
+@PlanningEntity(difficultyComparatorClass = TaskDifficultyComparator.class)
+@XStreamAlias("TaTask")
+public class Task extends TaskOrUser {
+
+    private long processInstanceId;
+    private String processId;
+    private String containerId;
+    private String name;
+    private int priority;
+    private Map<String, Object> inputData;
+
+    @PlanningPin
+    private boolean pinned;
+    private boolean published;
+
+    private Set<OrganizationalEntity> potentialOwners = new HashSet<>();
+    private Set<TypedLabel> typedLabels = new HashSet<>();
+
+    /**
+     * Planning variable: changes during planning, between score calculations.
+     */
+    @PlanningVariable(valueRangeProviderRefs = {"userRange", "taskRange"},
+            graphType = PlanningVariableGraphType.CHAINED)
+    private TaskOrUser previousTaskOrUser;
+
+    /**
+     * Shadow variable, let all Tasks have a reference to the anchor of the chain, the assigned user.
+     */
+    @AnchorShadowVariable(sourceVariableName = "previousTaskOrUser")
+    private User user;
+
+    /**
+     * When the previousTask changes we need to update the startTime for current task and also the startTime
+     * for all the tasks that comes after.  previousTask -> currentTask -> C -> D -> E since each task can only start
+     * after his previous one has finished. As part of the update the endTime for the modified tasks will also be updated.
+     */
+    @CustomShadowVariable(variableListenerClass = StartAndEndTimeUpdatingVariableListener.class,
+            // Arguable, to adhere to API specs (although this works), nextTask and user should also be a source,
+            // because this shadow must be triggered after nextTask and user (but there is no need to be triggered by those)
+            sources = {@PlanningVariableReference(variableName = "previousTaskOrUser")})
+    private Integer startTime; // In minutes
```

Comment:
hmmm, I don't like to bind the "unit" to the variable name

Reply:
The point is that if the variable name is descriptive enough, there is no need for the comment, which brings following benefits:
- meaning of the variable is clear not only at the declaration, but anywhere it's used
- comments often become obsolete as the code evolves - because with each relevant change, someone has to think about updating the comment too

Reply:
Also, there are several uses of time in various classes in this PR. Several methods in several classes. Adding the unit to the name of these methods will make it clear to the user that, if I have `Task.getEndTime()` and `TaskOrUser.getEndTime()`, that it is the same unit and that I can perform the same operations on them.

Reply:
I don't see this comment being addressed. I think the consistency across multiple classes is important.